lib/neo4j/shared/enum.rb

```diff
@@ -0,0 +1,120 @@
+require 'active_support/core_ext/hash/except'
+require 'active_support/core_ext/hash/slice'
+
+module Neo4j::Shared
+  module Enum
+    extend ActiveSupport::Concern
+
+    class ConflictingEnumMethodError < Neo4j::Error; end
+
+    module ClassMethods
+      attr_reader :neo4j_enum_data
+
+      def enum(parameters = {})
+        options, parameters = *split_options_and_parameters(parameters)
```

Comment:
Ooooh, I see what it's doing now.  This is so that you can specify multiple enums in the same definition, right?

I do like the second syntax a lot better, though...  @subvertallchris are you fine with breaking convention?  It would make the code a lot simpler, and I even think that the API would be simpler for people to use (even if it would be different than what people used to ActiveRecord would expect)